packages/core/src/agent/MessageReceiver.ts

```diff
@@ -275,11 +301,34 @@ export class MessageReceiver {
       },
     })
     problemReportMessage.setThread({
-      threadId: plaintextMessage['@id'],
+      threadId: plaintextMessage['@id'] as string,
     })
-    const outboundMessage = createOutboundMessage(connection, problemReportMessage)
+    const outboundMessage = createOutboundDIDCommV1Message(connection, problemReportMessage)
     if (outboundMessage) {
-      await this.messageSender.sendMessage(agentContext, outboundMessage)
+      await this.messageSender.sendDIDCommV1Message(agentContext, outboundMessage)
```

Comment:
I strongly agree on the need of removing complexity on current MessageSender class, but I'm not sure if it's better to define completely separated senders for each DIDComm version or to use a base MessageSender that receives the `OutboundMessageContext` object (where we can pass a V1/V2/Vx message) and according to it it would use one or another envelope.

I'm thinking mostly on sharing the logic for handling errors, adding messages to queues/sending to transports, etc. Of course this could also be achieved using an abstract MessageSender class and make V1MessageSender and V2MessageSender subclasses of it.